compose/material3/material3-window-size-class/build.gradle

```diff
@@ -111,6 +112,27 @@ androidXMultiplatform {
 
             }
         }
+        wasmJsMain {
+            dependencies {
+                implementation(libs.kotlinStdlib)
+            }
+        }
+
+        skikoMain.dependsOn(commonMain)
+        desktopMain.dependsOn(skikoMain)
+        nonJvmMain.dependsOn(skikoMain)
+        webMain.dependsOn(nonJvmMain)
+        jsMain.dependsOn(webMain)
+        wasmJsMain.dependsOn(webMain)
+        nativeMain.dependsOn(nonJvmMain)
+
+        skikoTest.dependsOn(commonTest)
+        desktopTest.dependsOn(skikoTest)
+        nonJvmTest.dependsOn(skikoTest)
+        webTest.dependsOn(nonJvmTest)
+        jsTest.dependsOn(webTest)
+        wasmJsTest.dependsOn(webTest)
+        nativeTest.dependsOn(nonJvmTest)
```

Comment:
```suggestion
        skikoMain.dependsOn(commonMain)
        targets.configureEach { target ->
            if (target.platformType !in [KotlinPlatformType.androidJvm, KotlinPlatformType.common]) {
                target.compilations["main"].defaultSourceSet {
                    dependsOn(skikoMain)
                }
                target.compilations["test"].defaultSourceSet {
                    dependsOn(commonTest)
                }
            }
        }
```

Reply:
I don't see a profit of this code. it is harder to understand and spends same number of lines of code

Reply:
Because you need to manually add dependencies to `nativeMain` to all specific architectures/OSes

Reply:
I wouldn't say I agree but I will make changes.
To have a flexible and common logic for all modules we need to extract it to the plugin. At the moment I see just copy-pasted code snippets around the project.

Reply:
your snippet adds every leaf sources as dependy to the skiko. why? we need hierarhical structure
![Uploading image.png…]()